src/components/app/FilteredResults.tsx

```diff
@@ -52,15 +36,22 @@ export const FilteredResults: React.FunctionComponent<IProps> = ({ className })
         />
         <SearchBar
           query={query}
-          onSearch={(query) => update({ query })}
+          onSearch={(newValue) => update({ query: newValue })}
           statusesWithScenarios={statusesWithScenarios}
           hideStatuses={hideStatuses}
-          onFilter={(hideStatuses) => update({ hideStatuses })}
+          onFilter={(newValue) => update({ hideStatuses: newValue })}
         />
       </div>
 
-      {filtered.length > 0 && <GherkinDocumentList gherkinDocuments={filtered} preExpand={true} />}
-      {filtered.length < 1 && <NoMatchResult query={query} />}
+      {filtered !== undefined && (
```

Comment:
`filtered` can be undefined if the (async) search initialisation hasn't happened yet - in which case we just won't render anything yet in this area.